engine/src/main/java/io/camunda/zeebe/engine/api/TaskResultBuilder.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.api;
+
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.RecordValue;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.Intent;
+
+/** Builder to compose the processing result */
+public interface TaskResultBuilder {
+
+  /**
+   * Appends a record to the result
+   *
+   * @return returns itself for method chaining
+   */
+  TaskResultBuilder appendRecord(
+      final long key,
+      final RecordType type,
+      final Intent intent,
+      final RejectionType rejectionType,
+      final String rejectionReason,
```

Comment:
:wrench: I feel that we don't need them here. I think they will always set to default values, since we will not write rejections here.